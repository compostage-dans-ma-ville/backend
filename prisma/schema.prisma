// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

generator dbml {
  provider = "prisma-dbml-generator"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id               Int      @id @default(autoincrement())
  email            String   @unique
  createdAt        DateTime @default(now())
  updatedAt        DateTime @default(now()) @updatedAt
  firstname        String
  lastname         String
  description      String?
  password         String
  isEmailConfirmed Boolean  @default(false)
  avatar           Image?   @relation(fields: [imageId], references: [id])
  imageId          String?
  role             UserRole @default(USER)

  organizations UserOrganizationRelation[]
  sites         UserSiteRelation[]
}

enum UserRole {
  USER
  OWNER
  ADMIN
}

enum SiteType {
  ADMINISTRATIVE_INSTITUTION
  BUILDING_FOOT
  COMPANY
  EDUCATIONAL_INSTITUTION
  SHARED
}

model Site {
  id               Int                 @id @default(autoincrement())
  createdAt        DateTime            @default(now())
  updatedAt        DateTime            @updatedAt
  launchDate       DateTime?
  name             String
  isPublic         Boolean             @default(true)
  type             SiteType
  treatedWaste     Int?
  description      String?
  accessConditions String?
  images           ImageSiteRelation[]
  dailySchedules   DailySchedule[]
  address          Address             @relation(fields: [addressId], references: [id], onDelete: Cascade)
  addressId        Int
  organization     Organization?       @relation(fields: [organizationId], references: [id])
  organizationId   Int?
  members          UserSiteRelation[]
}

model ImageSiteRelation {
  site   Site @relation(fields: [siteId], references: [id], onDelete: Cascade)
  siteId Int

  order Int

  image   Image  @relation(fields: [imageId], references: [id])
  imageId String

  @@id([siteId, imageId])
  @@unique([siteId, order])
}

model Image {
  id                String              @id
  mimeType          String
  imageSiteRelation ImageSiteRelation[]
  User              User[]
  Organization      Organization[]
}

// TODO: better modelization of the street number (1bis, 47Ter, ...)
// TODO: check if there are better coordinate type (distance between 2 coordinates) (small prio)
model Address {
  id          Int    @id @default(autoincrement())
  houseNumber String
  streetName  String
  zipCode     Int
  city        String
  latitude    Float
  longitude   Float

  site Site[]
}

model Organization {
  id          Int      @id @default(autoincrement())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  name        String   @unique
  description String?
  avatar      Image?   @relation(fields: [imageId], references: [id])
  imageId     String?
  links       String[]

  sites   Site[]
  members UserOrganizationRelation[]
}

enum OrganizationRole {
  MEMBER
  ADMIN
}

enum SiteRole {
  MEMBER
  ADMIN
  REFEREE
}

model UserOrganizationRelation {
  user   User @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId Int

  role OrganizationRole

  organization   Organization @relation(fields: [organizationId], references: [id], onDelete: Cascade)
  organizationId Int

  @@id([userId, organizationId])
}

model UserSiteRelation {
  user   User @relation(fields: [userId], references: [id])
  userId Int

  role SiteRole

  site   Site @relation(fields: [siteId], references: [id])
  siteId Int

  @@id([userId, siteId])
}

model DailySchedule {
  site   Site @relation(fields: [siteId], references: [id], onDelete: Cascade)
  siteId Int

  // between 0 and 6. 0 is Monday
  dayOfWeek Int

  openings Opening[]

  @@id([siteId, dayOfWeek])
}

model Opening {
  // between 0 and the amount of minutes in a day (1440)
  open  Int
  // between open and the amount of minutes in a day (1440)
  // close must be > open
  close Int

  DailySchedule          DailySchedule @relation(fields: [DailyScheduleSiteId, DailyScheduleDayOfWeek], references: [siteId, dayOfWeek], onDelete: Cascade)
  DailyScheduleSiteId    Int
  DailyScheduleDayOfWeek Int

  @@id([DailyScheduleSiteId, DailyScheduleDayOfWeek, open])
}
