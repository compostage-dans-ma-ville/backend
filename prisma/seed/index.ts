/* eslint-disable no-await-in-loop */
/* eslint-disable no-console */
import { Prisma, PrismaClient } from '@prisma/client'
import { faker } from '@faker-js/faker/locale/fr'
import * as dotenv from 'dotenv'

const prisma = new PrismaClient()

const fakerUser = (): Prisma.UserCreateInput => ({
  firstname: faker.name.firstName(),
  lastname: faker.name.lastName(),
  email: faker.internet.email(),
  password: faker.internet.password()
})

const seedUser = async (): Promise<void> => {
  const fakerRounds = 10
  for (let i = 0; i < fakerRounds; i++) {
    await prisma.user.create({ data: fakerUser() })
  }
}

const main = async (): Promise<void> => {
  dotenv.config()
  console.log('Seeding...')
  await seedUser()
}

main()
  .then(() => prisma.$disconnect())
  .catch(async (e) => {
    console.error(e)
    await prisma.$disconnect()
    process.exit(1)
  })
