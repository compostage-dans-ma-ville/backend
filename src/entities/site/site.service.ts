import { Injectable } from '@nestjs/common'
import type { CreateSiteDto } from './dto/create-site.dto'
import type { UpdateSiteDto } from './dto/update-site.dto'
import { PrismaService } from '~/prisma/prisma.service'
import { Prisma, Site } from '@prisma/client'

@Injectable()
export class SiteService {
  constructor(private readonly prisma: PrismaService) {}

  create(createSiteDto: CreateSiteDto) {
    return 'This action adds a new site'
  }

  findAll({ skip, take }: Prisma.SiteFindManyArgs): Promise<Site[]> {
    return this.prisma.site.findMany({
      skip,
      take
    })
  }

  count(): Promise<number> {
    return this.prisma.site.count()
  }

  findOne(id: number) {
    return `This action returns a #${id} site`
  }

  update(id: number, updateSiteDto: UpdateSiteDto) {
    return `This action updates a #${id} site`
  }

  remove(id: number) {
    return `This action removes a #${id} site`
  }
}
