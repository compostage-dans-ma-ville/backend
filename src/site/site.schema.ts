import { Address, Site } from '@prisma/client'
import { Factory } from 'nestjs-seeder'

export class SiteSchema implements Site {
  id: number

  createdAt: Date

  updatedAt: Date

  @Factory(faker => faker?.animal.rabbit())
    name: string

  @Factory(faker => faker?.lorem.paragraphs(5))
    description: string | null

  avatar: string | null

  @Factory(faker => ({
    houseNumber: faker?.address.buildingNumber(),
    streetName: faker?.address.street(),
    zipCode: Number(faker?.address.zipCode('#####')),
    city: faker?.address.cityName(),
    longitude: faker?.address.longitude(),
    latitude: faker?.address.latitude()
  }))
    address: Address

  addressId: number

  organizationId: number | null
}
