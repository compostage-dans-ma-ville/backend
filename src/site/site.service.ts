import { Injectable } from '@nestjs/common'
// import type { CreateSiteDto } from './dto/create-site.dto'
// import type { UpdateSiteDto } from './dto/update-site.dto'
import { PrismaService } from '~/prisma/prisma.service'
import {
  Address, ImageSiteRelation, Organization, Prisma, Schedule, Site
} from '@prisma/client'

@Injectable()
export class SiteService {
  constructor(private readonly prisma: PrismaService) {}

  // create(createSiteDto: CreateSiteDto) {
  //   return 'This action adds a new site'
  // }

  findAll({ skip, take }: Prisma.SiteFindManyArgs): Promise<Site[]> {
    return this.prisma.site.findMany({
      skip,
      take
    })
  }

  count(): Promise<number> {
    return this.prisma.site.count()
  }

  findOne(id: number): Prisma.Prisma__SiteClient<(Site & {
    Schedules: Schedule[];
    Images: ImageSiteRelation[];
    Address: Address;
    Organization: Organization | null;
}) | null, null> {
    return this.prisma.site.findUnique({
      include: {
        Schedules: true, Images: true, Address: true, Organization: true
      },
      where: { id }
    })
  }

  // update(id: number, updateSiteDto: UpdateSiteDto) {
  //   return `This action updates a #${id} site`
  // }

  remove(id: number): Prisma.Prisma__SiteClient<Site> {
    return this.prisma.site.delete({ where: { id } })
  }
}
