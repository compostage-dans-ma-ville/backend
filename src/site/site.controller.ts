import {
  Controller, Get, Param, Delete, Query, Req, ParseIntPipe
} from '@nestjs/common'
import type { Request } from 'express'
import { SiteService } from './site.service'
import { Site } from '.prisma/client'
import { ItemsQueryPipe } from '~/api-services/pagination/pipes/ItemsQueryPipe'
import { PageQueryPipe } from '~/api-services/pagination/pipes/PageQueryPipe'
import { PaginatedData } from '~/api-services/pagination/dto/PaginationData'
import { createPaginationData } from '~/api-services/pagination/creator/createPaginationData'
import { Prisma } from '@prisma/client'
import { ApiExtraModels, ApiTags } from '@nestjs/swagger'
import { ApiPaginatedResponse } from '~/api-services/pagination/ApiPaginationResponse'
import { GetSiteDto } from './dto/get-site.dto'

@Controller('sites')
@ApiTags('sites')
@ApiExtraModels(PaginatedData)
export class SiteController {
  constructor(private readonly siteService: SiteService) {}

  // @Post()
  // create(@Body() createSiteDto: CreateSiteDto) {
  //   return this.siteService.create(createSiteDto)
  // }

  @Get()
  @ApiPaginatedResponse(GetSiteDto)
  async findAll(
    @Req() req: Request,
    @Query('items', ItemsQueryPipe) items: number,
    @Query('page', PageQueryPipe) page: number,
  ): Promise<PaginatedData<Site>> {
    const sites = await this.siteService.findAll({
      skip: page * items,
      take: items
    })
    const totalItemCount = await this.siteService.count()

    return createPaginationData<Site>({
      url: `${req.protocol}://${req.get('Host')}${req.url}`,
      items: sites,
      queryOptions: { items, page },
      totalItemCount
    })
  }

  @Get(':id')
  findOne(@Param('id', ParseIntPipe) id: number): ReturnType<SiteService['findOne']> {
    return this.siteService.findOne(id)
  }

  // @Patch(':id')
  // update(@Param('id') id: string, @Body() updateSiteDto: UpdateSiteDto) {
  //   return this.siteService.update(+id, updateSiteDto)
  // }

  @Delete(':id')
  remove(@Param('id', ParseIntPipe) id: number): Prisma.Prisma__SiteClient<Site> {
    return this.siteService.remove(id)
  }
}
