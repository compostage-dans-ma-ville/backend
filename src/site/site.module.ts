import { Module } from '@nestjs/common'
import { SiteService } from './site.service'
import { SiteController } from './site.controller'
import { PrismaService } from '~/prisma/prisma.service'
import { DailyScheduleModule } from '~/dailySchedule/DailySchedule.module'

@Module({
  imports: [DailyScheduleModule],
  controllers: [SiteController],
  providers: [SiteService, PrismaService]
})
export class SiteModule {}
