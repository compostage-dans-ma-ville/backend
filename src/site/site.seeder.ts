import { Injectable } from '@nestjs/common'
import { Prisma } from '@prisma/client'
import { Seeder, DataFactory } from 'nestjs-seeder'
import { PrismaService } from '../prisma/prisma.service'
import { SiteSchema } from './site.schema'

@Injectable()
export class SiteSeeder implements Seeder {
  constructor(private prisma: PrismaService) {}

  async seed(): Promise<void> {
    const sites = DataFactory.createForClass(SiteSchema)
      .generate(10) as Prisma.SiteCreateManyInput[]
    await this.prisma.site.createMany({ data: sites })
  }

  async drop(): Promise<void> {
    await this.prisma.site.deleteMany()
  }
}
