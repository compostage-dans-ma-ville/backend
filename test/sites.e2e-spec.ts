import { Test, TestingModule } from '@nestjs/testing'
import { INestApplication } from '@nestjs/common'
import * as request from 'supertest'
import { SiteModule } from '~/site/site.module'
import { PrismaService } from '~/prisma/prisma.service'

describe('GET /sites', () => {
  let app: INestApplication
  let prisma: PrismaService

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [SiteModule]
    }).compile()

    app = moduleFixture.createNestApplication()
    prisma = app.get<PrismaService>(PrismaService)
    await app.init()
  })

  it('returns some sites', async () => {
    const { status, body } = await request(app.getHttpServer()).get('/sites')

    expect(status).toBe(200)
    expect(body).toBeDefined()
    expect(body).toMatchSnapshot()
  })
})
