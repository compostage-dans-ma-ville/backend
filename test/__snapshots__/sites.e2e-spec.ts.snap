// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /sites returns some sites 1`] = `
[
  {
    "addressId": 69,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": null,
    "id": 1,
    "name": "Californian",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 61,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Error laborum cum ab. Enim natus tempore aspernatur cupiditate culpa doloribus. Reiciendis sit quisquam corrupti non eveniet eveniet. Sed odit in doloribus. Reprehenderit blanditiis voluptas vero sapiente distinctio ipsum fugiat enim enim.
Eius ab et. Quasi omnis ipsa veniam unde vitae nesciunt amet consequuntur. Amet non corporis eum quidem dolorem recusandae. Error odio saepe ut excepturi veritatis saepe cum laboriosam. Necessitatibus laboriosam magnam quasi qui recusandae aliquam odio placeat. Sit doloribus officiis sapiente incidunt fugiat esse minus.
Sint eos soluta. Repellat corrupti voluptatibus iure quia excepturi sequi sapiente. Amet ex nemo iste eos tempora laborum quibusdam inventore atque.
Autem exercitationem eos cum distinctio consectetur laboriosam. Ipsum rem autem ut ipsa a sint officia mollitia. Corporis vel nemo laudantium. Dolor molestiae dolorem.
Asperiores quod perspiciatis veniam odit impedit eligendi pariatur non facilis. Autem odit asperiores exercitationem. Provident minima consequatur facilis quibusdam nesciunt sequi. Ex perspiciatis odio at ipsam aperiam. Assumenda in nostrum eveniet consequatur.",
    "id": 2,
    "name": "Lionhead",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 66,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Quia neque nihil aspernatur. Consectetur adipisci accusamus laborum nam blanditiis dolorum aut. Nemo itaque harum dicta id alias.
Adipisci assumenda aut neque. Voluptate alias amet maxime iure aliquam. Repellat blanditiis ea excepturi magnam officia. Deserunt animi atque magnam temporibus debitis tempora eum. Corporis numquam libero maxime distinctio ex dolor ipsa commodi repellat. Exercitationem dolores distinctio.
Odio nihil vitae debitis exercitationem repellat. Doloribus doloribus nihil. Excepturi rem perferendis reprehenderit. Error est in totam.
Illum necessitatibus quibusdam. Expedita possimus quaerat illo quae quis quidem consequatur. Voluptate nisi qui libero vel recusandae tempore omnis. Ratione facilis quibusdam. Atque veritatis id ab laudantium est. Placeat modi ex a commodi minima fugiat.
Molestiae reiciendis ex. Impedit non ipsam asperiores fugiat alias labore dolorum expedita rem. Tempore dolorum molestias. Ipsum necessitatibus officia saepe incidunt excepturi debitis error. Blanditiis dicta adipisci vero ullam.",
    "id": 3,
    "name": "English Spot",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 65,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Iste ducimus deleniti amet porro quod harum. Impedit rem deserunt sequi placeat. Excepturi doloremque est quibusdam tempore vero ipsum voluptatum. Quod itaque ipsam ducimus sint sunt dolore assumenda adipisci.
Reprehenderit illo veritatis veniam impedit veniam ipsum illo placeat tempore. Recusandae explicabo voluptatibus libero vitae provident distinctio accusamus. Nesciunt vel voluptas amet natus accusamus.
Assumenda nostrum rem culpa magni autem. Quae sapiente rerum molestiae omnis expedita. Delectus consequuntur quod reprehenderit tempore provident adipisci illo deleniti. Fugiat corrupti fugiat eaque illum facilis enim. Voluptatem molestiae tempore fuga saepe quam illo. Magni aspernatur optio doloribus ut eius assumenda nulla autem ut.
Molestias atque blanditiis id sequi necessitatibus animi qui ducimus. Nam molestias sed tempora. Possimus mollitia praesentium necessitatibus.
Et laboriosam ex deleniti explicabo explicabo ad debitis delectus et. Inventore quis dolore impedit rem necessitatibus omnis quibusdam explicabo. Quod libero deleniti tenetur animi voluptatum sequi. Error quas reiciendis. Eos qui deleniti. Dolore repellat molestias.",
    "id": 4,
    "name": "Blanc de Hotot",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 67,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Recusandae tempore sequi esse. Ducimus ex soluta beatae. Quasi odit facere ex ex aspernatur. Unde repellat veritatis corrupti aliquam qui dignissimos omnis.
Incidunt temporibus labore aspernatur quisquam vero nesciunt. Nihil a ullam. Pariatur veniam culpa. Facilis dolorem omnis qui eaque alias.
Repellendus modi fugit dolore rem sint explicabo quidem voluptates. Hic cum est nihil quia sapiente minima aliquid. Temporibus architecto cum alias dolorum porro. Cumque ducimus consectetur voluptatum quaerat dolorum eveniet iure minima cum.
Harum vero sapiente molestiae iste voluptas nostrum iste velit. Quasi totam fugiat. Illo quae hic nam hic. Non tempora consequatur dicta placeat nostrum illo.
Eos ea placeat expedita adipisci quae in esse eum. Hic quasi quisquam. Ullam deserunt necessitatibus odit sequi praesentium adipisci minima unde. Itaque eveniet dolores.",
    "id": 5,
    "name": "Netherland Dwarf",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 63,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Temporibus alias laudantium unde pariatur deserunt. Magni et quos. Aliquid ipsum dolore voluptates.
Magni sunt quisquam perspiciatis deserunt porro a ad at fugiat. Omnis eligendi quaerat enim. Magnam iste accusamus recusandae consequuntur enim nihil.
Voluptatum mollitia ab dolore dolor enim nulla vero dolorem architecto. Dolore quos et laborum quasi dolorem distinctio veritatis. Architecto sed ipsum fuga voluptatibus.
Voluptatum fuga expedita quaerat. Minus ea tempora. Sunt cupiditate debitis magnam voluptatum culpa maxime quisquam. Ab deserunt cum quaerat voluptas consequatur impedit. Expedita distinctio doloremque minima aspernatur. Nulla aspernatur quibusdam aspernatur omnis accusantium.
Repellat architecto et porro. Alias omnis explicabo sunt labore sunt. Ad in maiores atque quibusdam sit. Ducimus harum eveniet laudantium accusamus sunt aliquid.",
    "id": 6,
    "name": "Standard Chinchilla",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 64,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": null,
    "id": 7,
    "name": "Tan",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 62,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": null,
    "id": 8,
    "name": "Beveren",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 70,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": null,
    "id": 9,
    "name": "Giant Angora",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 68,
    "avatar": null,
    "createdAt": "2023-02-12T19:20:21.451Z",
    "description": "Cumque laboriosam dignissimos illum porro inventore cumque commodi saepe. Vero non expedita repudiandae nisi quibusdam cum maiores. Sequi quos distinctio atque odit harum itaque aspernatur. Eligendi odio qui eius exercitationem voluptates officia nam porro hic. Repellendus rerum adipisci quas. Reprehenderit reprehenderit neque aperiam labore aut error libero.
Id dolorem rerum a earum. Doloremque molestias voluptates quibusdam et alias aliquid facilis quam optio. Ea esse impedit dicta quibusdam.
Pariatur asperiores numquam magni iure quia vitae aliquam laboriosam sed. Doloribus dolores quasi voluptate. Officia asperiores ea consectetur tenetur. Dolor esse dolorum eius perferendis temporibus exercitationem illum sit. Aperiam ea modi consectetur debitis nam sint.
Illum culpa aspernatur atque iusto aut accusantium quae mollitia. Veniam repudiandae inventore et provident inventore vitae doloremque officiis aliquid. Doloremque neque assumenda similique quae cumque ratione. Placeat repudiandae ex. Mollitia odio voluptates perferendis voluptatibus qui laboriosam cupiditate. Magni ab qui eligendi labore vitae officia sequi dolor.
Ipsum eveniet magnam vel sit soluta nisi. Autem quibusdam maxime. Ab aspernatur corporis dignissimos optio quod laudantium.",
    "id": 10,
    "name": "Mini Satin",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:20:21.451Z",
  },
  {
    "addressId": 88,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": "A recusandae fugiat consectetur tempora odit eligendi quia. Dolore quae quidem. Blanditiis modi est itaque. Ut dicta voluptas quisquam voluptatum quae perferendis optio quaerat dolorum. Numquam hic vero qui necessitatibus eos quam itaque iste cumque.
Eum rem ea dicta quaerat cupiditate. Nulla inventore fuga culpa vero exercitationem vero exercitationem. Porro laboriosam repellendus vero ea eveniet. Sit cupiditate ea quos repudiandae vitae est. Porro architecto nobis aperiam deleniti tempora molestiae quisquam ea. Qui laudantium praesentium.
Earum nesciunt aliquam voluptatibus aliquid suscipit eum quo. Exercitationem accusantium fugiat rerum. Veritatis ullam a animi. Illum sint vero amet. Id odit veniam recusandae fugit autem in.
Possimus impedit iure quam ut quia voluptas. Deleniti ab architecto voluptas. Aliquid quas amet eum voluptatibus.
In assumenda officiis. Quam sapiente enim nemo. Repellat quaerat sunt voluptate quam. Iure iure perferendis occaecati. Eaque dicta molestias.",
    "id": 11,
    "name": "Lilac",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 89,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": null,
    "id": 12,
    "name": "French Angora",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 82,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": "Dignissimos sapiente cum mollitia dolore. Voluptatem tenetur minima expedita amet tenetur. Dolores dolorem nulla unde corrupti quo dolorem rerum unde numquam. Dolore iure quaerat illum.
Ducimus aliquid modi consequatur sequi doloribus aut. Pariatur minus perspiciatis commodi beatae. Aliquam provident nobis perspiciatis. Ipsa earum ipsa ab debitis eaque cumque provident. Repellat dignissimos quisquam eaque natus quaerat. Earum doloremque accusantium incidunt deleniti ab explicabo voluptatem ut voluptatum.
Fugit totam ducimus nostrum ratione. Reprehenderit consequatur perferendis quod autem debitis dolorem. Necessitatibus sint ex commodi qui commodi. Corporis atque facilis sed sapiente saepe architecto similique quae aliquid. Debitis deserunt et exercitationem quasi doloremque eligendi aspernatur. Ipsam tempore eum omnis dolorum impedit quis sed quae placeat.
Maxime repellendus optio temporibus cumque occaecati. Et est nam dolor maxime unde iure. Aspernatur excepturi occaecati molestiae excepturi. Veniam quia deserunt nihil laudantium non quod.
Dolores veniam eveniet repudiandae molestiae. Doloremque distinctio expedita vitae ad expedita atque ratione saepe nam. Dolore nulla voluptate ratione. Quasi ullam omnis repellendus est. Quod occaecati velit illo molestiae doloremque porro eius officia sapiente. Atque enim qui.",
    "id": 13,
    "name": "English Angora",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 86,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": null,
    "id": 14,
    "name": "Rex",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 87,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": "Mollitia ipsa minima nostrum excepturi quae iusto modi laborum incidunt. Voluptate atque sed maiores. Asperiores consequatur accusamus vel totam neque soluta ex reiciendis. Doloremque quis illum tenetur alias sunt rem nisi ea. Quam doloribus earum minima sunt porro.
Aliquam distinctio numquam neque fugiat quae nisi aperiam ex tempore. Nesciunt iure natus sunt unde. Accusamus explicabo cupiditate.
Rem atque tempora vel. Ipsum odit sequi reiciendis perferendis totam quisquam. Repellat voluptas pariatur animi ipsam explicabo aperiam veritatis molestias corrupti. Perspiciatis quis id veniam a cum. Odio numquam magnam facilis numquam. Maiores magni eius autem.
Illo similique atque unde voluptas sequi veniam laborum. Nostrum quas id quam repellendus quisquam architecto. Sint voluptas alias.
Inventore voluptas officia quasi. Quaerat atque perspiciatis magnam at. Nihil accusamus ab nam ipsa saepe odio non expedita repudiandae. Unde atque dicta.",
    "id": 15,
    "name": "Champagne D’Argent",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 81,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": null,
    "id": 16,
    "name": "American Chinchilla",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 90,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": "Minus mollitia doloremque dolorem dolorum. Optio eligendi sapiente neque optio modi enim. Voluptatum minus accusamus totam consequuntur animi placeat fugit molestiae alias. Ipsum itaque aspernatur. Repudiandae repellendus laudantium odio reprehenderit molestiae eveniet. Eos iure a laborum.
Eveniet aut perferendis omnis consequatur quasi recusandae. Illum totam praesentium laboriosam. Quo blanditiis illo.
Quas cumque quidem veniam amet. Ipsam laudantium maiores perferendis distinctio. Cumque molestiae quae. Reiciendis natus molestias atque inventore consectetur dolor. Voluptatum id distinctio a rem quos officia maxime ducimus.
Minus nihil nihil iure magnam. Quod facere rem ipsa sit nobis laboriosam debitis harum eligendi. Provident necessitatibus dolore ipsum facere reiciendis. Nulla harum incidunt voluptates aliquam. Accusantium ullam consequuntur qui culpa at.
Quisquam maiores fugit tempora dicta incidunt. Architecto similique velit nulla aliquam laboriosam soluta exercitationem. Eveniet veritatis odio quaerat adipisci. Incidunt totam ea expedita optio voluptatibus nobis adipisci officia. Expedita sit mollitia odio. Sapiente doloremque molestiae deleniti quis laudantium quos facilis itaque ipsum.",
    "id": 17,
    "name": "Britannia Petite",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 85,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": null,
    "id": 18,
    "name": "French Angora",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 83,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": null,
    "id": 19,
    "name": "Rex",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
  {
    "addressId": 84,
    "avatar": null,
    "createdAt": "2023-02-12T19:24:30.773Z",
    "description": "Suscipit cupiditate temporibus laudantium beatae sint beatae itaque accusamus. Veritatis officia ipsum. Velit nobis aliquid laudantium molestiae aliquid porro neque minus. Neque tenetur nostrum nulla.
Placeat repellendus eius accusamus. Voluptas quis maxime consectetur molestias quos deserunt autem. Excepturi iure quis quasi commodi cumque.
In itaque dolores quasi inventore omnis. Aliquid aut quia nisi ratione aliquam. Minus vitae quam officia aspernatur.
Voluptatum non quibusdam iste omnis placeat nihil excepturi veritatis. Quaerat illo ipsum rerum nemo culpa ducimus. Veritatis occaecati asperiores corrupti suscipit suscipit laborum cupiditate ipsa nobis. Eaque maiores cumque numquam debitis odio velit tempora consequuntur. Accusamus quae quibusdam harum.
Iure architecto quod quo ullam tempora asperiores excepturi. Fugiat itaque fugiat vitae necessitatibus quaerat explicabo saepe nostrum nemo. Saepe impedit quia. Ipsum et fugit explicabo quasi minima inventore repellat. Laborum non aut commodi.",
    "id": 20,
    "name": "Satin",
    "organizationId": null,
    "updatedAt": "2023-02-12T19:24:30.773Z",
  },
]
`;
